src/solution.jl

```diff
@@ -50,3 +65,86 @@ function init_solution(instance::Instances)
+        end
+    end
+
+    # Dynamic (right to left)
+    for counter in 0:(nb-1)
+        index = nb-counter
+        car = solution.sequence[index]
+        for option in 1:nb_RC
+            # Has option -> next sequence + 1
+            if instance.RC_flag[car,option]
+                solution.M1[option,index] = solution.M1[option,index+1] + 1
+            else
+                solution.M1[option,index] = solution.M1[option,index+1]
+            end
+
+            # Is there one car not reach anymore ?
+            index_first_out = index + instance.RC_q[option]
+            if index_first_out <= nb
+                car_first_out = solution.sequence[index_first_out]
+                # It had option -> sequence - 1
+                if instance.RC_flag[car_first_out,option]
+                    solution.M1[option,index] = solution.M1[option,index] - 1
+                end
+            end
+        end
+    end
+
+    # Update M2 and M3 (left to right)
+    for option in 1:nb_RC
+        solution.M2[option, 1] = (solution.M1[option, 1] >  instance.RC_p[option] ? 1 : 0)
+        solution.M3[option, 1] = (solution.M1[option, 1] >= instance.RC_p[option] ? 1 : 0)
+        for index in 2:nb
+            solution.M2[option, index] = solution.M2[option, index-1] + (solution.M1[option, index] >  instance.RC_p[option] ? 1 : 0)
+            solution.M3[option, index] = solution.M3[option, index-1] + (solution.M1[option, index] >= instance.RC_p[option] ? 1 : 0)
+        end
+    end
+end
+
+"""
+    update_matrices_new_car!(solution::Solution, position::Int, instance::Instance)
+
+Update `solution.M1`, `solution.M2` and `solution.M3` for a new car add at a given position.
+DOES NOT update columns for sequences excluding the position. Assert made : the function is called
+when we add a car in an tail-less sequence.
+"""
+function update_matrices_new_car!(solution::Solution, position::Int, instance::Instance)
+    nb_RC = instance.nb_HPRC+instance.nb_LPRC
+    car = solution.sequence[position]
+
+    for option in 1:nb_RC
+        for index in (position - instance.RC_q[option] + 1):position
+            if index > 0
+                # new car has option -> update M1
+                if instance.RC_flag[car,option]
+                    solution.M1[option,index] = solution.M1[option,index] + 1
+                end
+
+                # Columns of M2 and M3 can be update
+                if index == 1
+                    solution.M2[option,index] = 0 + (solution.M1[option,index] >  instance.RC_p[option] ? 1 : 0)
+                    solution.M3[option,index] = 0 + (solution.M1[option,index] >= instance.RC_p[option] ? 1 : 0)
+                else
+                    solution.M2[option,index] = solution.M2[option,index-1] + (solution.M1[option,index] >  instance.RC_p[option] ? 1 : 0)
+                    solution.M3[option,index] = solution.M3[option,index-1] + (solution.M1[option,index] >= instance.RC_p[option] ? 1 : 0)
+                end
```

Comment:
Check the index in order to attribute a first value to `solution.M2[option,index]` and `solution.M3[option,index]`. Then use the comment on lines +114 to +115.

To be consistent in this function chose `[., .]` instead of `[.,.]`